lms/djangoapps/commerce/tests/test_signals.py

```diff
@@ -0,0 +1,117 @@
+"""
+Tests for signal handling in commerce djangoapp.
+"""
+from django.test import TestCase
+from django.test.utils import override_settings
+
+import mock
+from opaque_keys.edx.keys import CourseKey
+from student.models import UNENROLL_DONE
+from student.tests.factories import UserFactory, CourseEnrollmentFactory
+
+from commerce.signals import refund_seat
+from commerce.tests import TEST_API_URL, TEST_API_SIGNING_KEY
+from commerce.tests.mocks import mock_create_refund
+
+
+@override_settings(ECOMMERCE_API_URL=TEST_API_URL, ECOMMERCE_API_SIGNING_KEY=TEST_API_SIGNING_KEY)
+class TestRefundSignal(TestCase):
+    """
+    Exercises logic triggered by the UNENROLL_DONE signal.
+    """
+
+    def setUp(self):
+        super(TestRefundSignal, self).setUp()
+        self.requester = UserFactory(username="test-requester")
+        self.student = UserFactory(username="test-student")
+        self.course_enrollment = CourseEnrollmentFactory(
+            user=self.student,
+            course_id=CourseKey.from_string('course-v1:org+course+run'),
+        )
+        self.course_enrollment.refundable = mock.Mock(return_value=True)
+
+    def send_signal(self, skip_refund=False):
+        """
+        DRY helper: emit the UNENROLL_DONE signal, as is done in
+        common.djangoapps.student.models after a successful unenrollment.
+        """
+        UNENROLL_DONE.send(sender=None, course_enrollment=self.course_enrollment, skip_refund=skip_refund)
+
+    @override_settings(ECOMMERCE_API_URL=None, ECOMMERCE_API_SIGNING_KEY=None)
+    def test_no_service(self):
+        """
+        Ensure that the receiver quietly bypasses attempts to initiate
+        refunds when there is no external service configured.
+        """
+        with mock.patch('commerce.signals.refund_seat') as mock_refund_seat:
+            self.send_signal()
+            self.assertFalse(mock_refund_seat.called)
+
+    @mock.patch('commerce.signals.refund_seat')
+    def test_receiver(self, mock_refund_seat):
+        """
+        Ensure that the UNENROLL_DONE signal triggers correct calls to
+        refund_seat(), when it is appropriate to do so.
+
+        TODO (jsa): ideally we would assert that the signal receiver got wired
+        up independently of the import statement in this module.  I'm not aware
```

Comment:
thanks - the problem i'm musing over here is the opposite. [The import statement in __init__.py](https://github.com/edx/edx-platform/pull/8127/files#diff-f17399a99e052ec3802476d5a5508f15R31) takes care of wiring up the receiver, but if that line ever gets removed for some reason, these tests will still pass because this module imports commerce.signals directly.  I don't know why anyone would ever do that, but I don't think it's possible to validate with a unit test.   An acceptance or integration test would be needed to catch it.